### PR TITLE
[JUJU-4615] Ask user to create local share after installing snap

### PR DESCRIPTION
**Pull request by anvial**
This PR adds a warning that asks to create the .local/share directory manually. This warning should appear every time Juju fails directory existence checks (including installation). This would enhance usability during working with juju strictly confined snap.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- [x] Comments saying why design decisions were made
- [x] Go unit tests, with comments saying what you're testing

## QA steps

```sh
go test github.com/juju/juju/cmd/juju/commands/... -gocheck.v

snapcraft --use-lxd
sudo snap install <juju_snap_filename>.snap
juju clouds
```

## Links

**Launchpad bug:** https://bugs.launchpad.net/juju/+bug/2031483

**Jira card:** JUJU-4615

### CODE REVIEW

**Review by SimonRichardson — COMMENTED** [review]
I think we should actually move away from the home interface and to [personal-files-interface](https://snapcraft.io/docs/personal-files-interface) . You'll need to ask snap team for approval, but will solve this in a much better way.

**Review by SimonRichardson — CHANGES_REQUESTED** [review]
We'll be warning about false positives here, instead, we should return typed errors and only show the warning on the home dir issue.

--- 

Alternatively, we could split the home dir and ssh calls into two...

**Review by SimonRichardson — COMMENTED** [review]
This isn't correct either, as if you're running out side of a snap you'll get this warning.

**Review by SimonRichardson — APPROVED** [review]

File: cmd/juju/commands/main.go [hunk 1]
```diff
@@ -146,6 +146,9 @@ func (m jujuMain) Run(args []string) int {
 
 	if err = juju.InitJujuXDGDataHome(); err != nil {
 		cmd.WriteError(ctx.Stderr, err)
+		if errors.Is(err, errors.NotFound) {
+			ctx.Warningf("Installing juju from a snap, the .local/share directory should be manually created as it is strictly confined.")
```
Comment (manadart) [inline cmd/juju/commands/main.go:150]: A sentence like this, where you state 2 things at the front of the sentence, and then refer to "it" at the back, makes it ambiguous.

It should be something like:
"Installing Juju in a strictly confined Snap. To ensure correct operation, create the .local/share directory manually."

**Pull request merged**